modules/dokkatoo-plugin/src/main/kotlin/DokkatooBasePlugin.kt

```diff
@@ -98,7 +98,12 @@ constructor(
       publicationEnabled.convention(true)
       onlyIf("publication must be enabled") { publicationEnabled.getOrElse(true) }
 
-      generator.dokkaSourceSets.addAllLater(providers.provider { dokkatooExtension.dokkatooSourceSets })
+       generator.dokkaSourceSets.addAllLater(
+         providers.provider {
+           // exclude suppressed source sets to avoid unnecessary dependency resolution for them
+           dokkatooExtension.dokkatooSourceSets.filterNot { it.suppress.get() }
+         }
+       )
```

Comment:
Is there a difference between filtering the suppressed source sets here and in the existing location, in [DokkaSourceSetBuilder](https://github.com/adamko-dev/dokkatoo/blob/0850e6e609ef371986fcee6e6572de71d5f4b17a/modules/dokkatoo-plugin/src/main/kotlin/dokka/parameters/builders/DokkaSourceSetBuilder.kt#L28-L40)?

Reply:
intention is to avoid getting source sets as early as possible, before task executes, so that unnecessary compilation is avoided